sklearn/neighbors/approximate.py

```diff
@@ -309,15 +318,23 @@ def _get_radius_neighbors(self, query, max_depth, bin_queries, radius):
         total_neighbors = np.array([], dtype=int)
         total_distances = np.array([], dtype=float)
 
+        index_offsets = bin_queries // (2**self._residual)
+
         while (max_depth > self.min_hash_match and
                ratio_within_radius > threshold):
             left_mask = self._left_mask[max_depth]
             right_mask = self._right_mask[max_depth]
             candidates = []
             for i in range(self.n_estimators):
-                start, stop = _find_matching_indices(self.trees_[i],
+                start, stop = _find_matching_indices(self.trees_[i]
+                                                     [self._locations
+                                                      [i, index_offsets[i]]:
```

Comment:
For `depth < offset` (we need to come up with a better name than offset!), this range is too restrictive. That is, with `min_hash_match=4` and `offset=5` this will still be extracting the range for 5 matching bits, not for 4. It might be easiest if some of this logic is moved to `_find_matching_indices`.